frontend/src/metabase/common/components/EntityPicker/components/ItemList/RootItemList.tsx

```diff
@@ -0,0 +1,125 @@
+import { useMemo } from "react";
+import { t } from "ttag";
+
+import { useCollectionQuery } from "metabase/common/hooks";
+import { PERSONAL_COLLECTIONS } from "metabase/entities/collections";
+import { useSelector } from "metabase/lib/redux";
+import { getUser, getUserIsAdmin } from "metabase/selectors/user";
+import type { CollectionId } from "metabase-types/api";
+
+import type {
+  EntityPickerOptions,
+  CollectionPickerItem,
+  TypeWithModel,
+  TisFolder,
+} from "../../types";
+
+import { ItemList } from "./ItemList";
+
+const personalCollectionsRoot: CollectionPickerItem = {
+  ...PERSONAL_COLLECTIONS,
+  can_write: false,
+  model: "collection",
+  location: "/",
+  description: "",
+};
+
+interface RootItemListProps<CollectionPickerItem extends TypeWithModel> {
+  onClick: (val: CollectionPickerItem) => void;
+  selectedItem: CollectionPickerItem | null;
+  options: EntityPickerOptions;
+  isFolder: TisFolder<CollectionPickerItem>;
+  isCurrentLevel: boolean;
+}
+/**
+ * This is a special item list that exists "above" our analytics and might include:
+ * a) the highest-level collections the user can access (often "our analytics")
+ * b) the user's personal collection
+ * c) a top level folder including all personal collections (admin only)
+ */
+export const RootItemList = ({
+  onClick,
+  selectedItem,
+  options,
+  isFolder,
+  isCurrentLevel,
+}: RootItemListProps<CollectionPickerItem>) => {
+  const isAdmin = useSelector(getUserIsAdmin);
+  const currentUser = useSelector(getUser);
+
+  const { data: personalCollection, isLoading: isLoadingPersonalCollecton } =
```

Comment:
error state?

Reply:
this one is actually handled implicitly and silently in the `useMemo`